crates/environ/src/module.rs

```diff
@@ -249,9 +296,12 @@ impl ModuleTranslation<'_> {
         let mut offset = 0;
         for (memory, pages) in page_contents {
             let mut page_offsets = Vec::with_capacity(pages.len());
-            for (byte_offset, page) in pages {
+            for (page_index, page) in pages {
```

Comment:
Interesting, was this a bug in the original code? I see the insertion above `contents.entry(page_index)` so this is correct now, but seemed to be using a page index as a byte offset previously. Or was the first tuple element used as a page index elsewhere?

Reply:
I looked at a few other places but I think this was just a typo because everywhere else named and used this as a page index. I was a bit worried myself though and had to do a few double-takes as I updated this!